packages/account-sdk/src/interface/payment/getPaymentStatus.ts

```diff
@@ -130,9 +130,28 @@ export async function getPaymentStatus(options: PaymentStatusOptions): Promise<P
       if (txReceipt?.logs) {
         const network = testnet ? 'baseSepolia' : 'base';
         const usdcAddress = TOKENS.USDC.addresses[network].toLowerCase();
+        // Normalize sender address for comparison
+        const senderAddress: Address | undefined = receipt.result.sender
+          ? getAddress(receipt.result.sender)
+          : undefined;
 
-        for (const log of txReceipt.logs) {
-          if (log.address?.toLowerCase() === usdcAddress) {
+        // Collect all USDC transfers
+        const usdcTransfers: Array<{
+          from: string;
+          to: string;
+          value: bigint;
+          formattedAmount: string;
+          logIndex: number;
+        }> = [];
+
+        for (let i = 0; i < txReceipt.logs.length; i++) {
+          const log = txReceipt.logs[i];
+
+          // Check if this is a USDC log
+          const logAddressLower = log.address?.toLowerCase();
+          const isUsdcLog = logAddressLower === usdcAddress;
+
```

Comment:
nit: wonder if we should extract this part out to a helper, given this function is getting a bit long